pulsar-broker-auth-sasl/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderSasl.java

```diff
@@ -110,6 +113,8 @@ public void initialize(ServiceConfiguration config) throws IOException {
             throw new IllegalArgumentException(msg);
         }
         this.signer = new SaslRoleTokenSigner(secret);
+        this.authStates = Caffeine.newBuilder()
+                .expireAfterWrite(config.getInflightSaslContextExpiryMs(), TimeUnit.MILLISECONDS).build();
```

Comment:
This kind of caching would still be vulnerable to a high frequency of calls, similar to a TCP SYN flood. Should we also put an upper limit on the number of in flight contexts so that old contexts are dropped?